dmriprep/interfaces/images.py

```diff
@@ -91,55 +85,27 @@ class RescaleB0(SimpleInterface):
     output_spec = _RescaleB0OutputSpec
 
     def _run_interface(self, runtime):
+        from nipype.utils.filemanip import fname_presuffix
+
+        out_b0s = fname_presuffix(
+            self.inputs.in_file,
+            suffix='_rescaled',
+            use_ext=True,
+            newpath=str(Path(runtime.cwd).absolute())
+        )
+        out_ref = fname_presuffix(
+            self.inputs.in_file,
+            suffix='_ref',
+            use_ext=True,
+            newpath=str(Path(runtime.cwd).absolute())
+        )
+
         self._results['out_b0s'] = rescale_b0(
             self.inputs.in_file,
-            self.inputs.mask_file,
-            newpath=runtime.cwd
+            self.inputs.mask_file, out_b0s
```

Comment:
I think this is what introduced #90 - the mask and the b0s are swapped.

Reply:
it failed at the very first step where the b0s get extracted

Reply:
yup I realized after commenting. I'm still going through this - I might have found a solution.